### ISSUE TITLE
Xpath support

### ISSUE THREAD

**Issue by simPod** [chrome-php/headless-chromium-php#333]
Hi, what would you say to adding a support for xpath to the current css selectors?

We use js already to find elements using css selectors.

Looking for elements using xpath seems to be as easy as

```js
var xpath = "//a[text()='SearchingText']";
var matchingElement = document.evaluate(xpath, document, null, XPathResult.FIRST_ORDERED_NODE_TYPE, null).singleNodeValue;
```

I propose to add `Selector` interface with `CssSelector` and `XPathSelector` implementations, used as

`Mouse::find(Selector $selectors, int $position = 1): self`

in v2.0

**Comment by enricodias** [comment]
Couldn't we add this feature in v1 without breaking backwards compatibility?

**Comment by simPod** [comment]
I'm not good at not breaking BC :D 

We would have to introduce another method. But IMO it's better to have only one so we should target that at least in a long term

**Comment by enricodias** [comment]
How about implementing this in another method at first and just change the interface later on v2? The underlying code would be the same. People tend to not upgrade major versions too often. We could group several bc changes and release them together in v2, but have most of the features already implemented in v1.

I also don't like the idea of the `Mouse` class finding elements. When I implemented it I was thinking that it should act like a real user and not like a mouse. The lib was easy to use that way, but now I think we can improve on that.

**Comment by simPod** [comment]
Where would you place finding elements instead now?

**Comment by enricodias** [comment]
I don't have an idea yet 😅
We can think about this for v2.

**Comment by enricodias** [comment]
But your `Selector` class can be implemented in the meantime and used in another method.

**Comment by GrahamCampbell** [comment]
Implemented in v1.5.0.
